platform/darwin/src/MGLStyleValue_Private.h

```diff
@@ -13,6 +15,35 @@
 
 #include <array>
 
+template <typename MBGLType, typename MGLType>
+class MGLStyleEnumerationValueTransformer {
```

Comment:
Can this method live inside `MGLStyleValueTransformer`? Use optional template parameters on the class to allow both `NSValue *` and `MGLLineCap` to be specific, and use conditional template parameters on the enumeration-specific methods to enable the methods only for enumerations.